src/Task/Recurrence.ts

```diff
@@ -132,50 +132,56 @@ export class Recurrence {
-                }
-                if (this.scheduledDate) {
-                    const originalDifference = window.moment.duration(this.scheduledDate.diff(this.referenceDate));
+        if (next === null) {
+            return null;
+        }
 
-                    // Cloning so that original won't be manipulated:
-                    scheduledDate = window.moment(next);
-                    // Rounding days to handle cross daylight-savings-time recurrences.
-                    scheduledDate.add(Math.round(originalDifference.asDays()), 'days');
-                }
-                if (this.dueDate) {
-                    const originalDifference = window.moment.duration(this.dueDate.diff(this.referenceDate));
-
-                    // Cloning so that original won't be manipulated:
-                    dueDate = window.moment(next);
-                    // Rounding days to handle cross daylight-savings-time recurrences.
-                    dueDate.add(Math.round(originalDifference.asDays()), 'days');
-                }
-            }
+        // Keep the relative difference between the reference date and
+        // start/scheduled/due.
+        let startDate: Moment | null = null;
+        let scheduledDate: Moment | null = null;
+        let dueDate: Moment | null = null;
 
+        // Only if a reference date is given. A reference date will exist if at
+        // least one of the other dates is set.
+        if (this.referenceDate === null) {
             return {
-                startDate,
-                scheduledDate,
-                dueDate,
+                startDate: null,
+                scheduledDate: null,
+                dueDate: null,
             };
         }
 
-        return null;
+        if (this.startDate) {
+            const originalDifference = window.moment.duration(this.startDate.diff(this.referenceDate));
+
+            // Cloning so that original won't be manipulated:
+            startDate = window.moment(next);
+            // Rounding days to handle cross daylight-savings-time recurrences.
+            startDate.add(Math.round(originalDifference.asDays()), 'days');
+        }
+        if (this.scheduledDate) {
+            const originalDifference = window.moment.duration(this.scheduledDate.diff(this.referenceDate));
+
+            // Cloning so that original won't be manipulated:
+            scheduledDate = window.moment(next);
+            // Rounding days to handle cross daylight-savings-time recurrences.
+            scheduledDate.add(Math.round(originalDifference.asDays()), 'days');
+        }
+        if (this.dueDate) {
+            const originalDifference = window.moment.duration(this.dueDate.diff(this.referenceDate));
+
+            // Cloning so that original won't be manipulated:
+            dueDate = window.moment(next);
+            // Rounding days to handle cross daylight-savings-time recurrences.
+            dueDate.add(Math.round(originalDifference.asDays()), 'days');
+        }
```

Comment:
I will merge this as-is...

And I rather imagine this might be on your radar for a future improvement...

But I couldn't resist trying to refact and remove that duplication.

I ended up with:

```ts
        if (this.startDate) {
            startDate = this.nextFieldDate(this.startDate, next);
        }
        if (this.scheduledDate) {
            scheduledDate = this.nextFieldDate(this.scheduledDate, next);
        }
        if (this.dueDate) {
            dueDate = this.nextFieldDate(this.dueDate, next);
        }
```

And

```ts
    private nextFieldDate(currentDate: moment.Moment, next: Date) {
        const originalDifference = window.moment.duration(currentDate.diff(this.referenceDate));

        // Cloning so that original won't be manipulated:
        const newDate = window.moment(next);
        // Rounding days to handle cross daylight-savings-time recurrences.
        newDate.add(Math.round(originalDifference.asDays()), 'days');

        return newDate;
    }
```

Reply:
Yup, I went a bit farer than that, but the change was too big for one PR